compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRInstructionWriter.java

```diff
@@ -421,6 +422,27 @@ public void visit(BIROperand birOperand) {
         }
     }
 
+    public void visit(BIRArgument birArgument) {
```

Comment:
This changes the BIR format right? we'll need to update the kaitai file ?
@Kishanthan wdyt?

Reply:
Yes, the kaitai file has to be updated.

Reply:
This should fail the BIR tests also I believe.

Reply:
@chiranSachintha , if this method does the same as visit(BIROperand birOperand), then you can call the same than duplicating the code.

Reply:
Yes. I will fix it.

Reply:
Fixed. Do we need to update the `kaitai file`?(When using `visit(BIROperand birOperand)` method for BIRArgument).

Reply:
Btw, do we need this method at all? I don't see that you added a accept method for BIRArgument and you are not doing anything differently for BIRArgument. So It should still work without this method I think.

Reply:
Let me step back a bit. I think we need to have a separate write method for BIRArgument. Because, BIRArgument has a state called ArgumentState and it is not captured in the BIR. So, in BIR reader(s), like BIRSymbolEnter and nBallerina readers, we will not have this information to propagate to the backend. 

For the moment, we can go ahead with merging this. But lets capture this issue @MaryamZi @chiranSachintha

Reply:
Thanks. I've created https://github.com/ballerina-platform/ballerina-lang/issues/28058 to track fixing the suggestions to this PR.